sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -116,7 +121,7 @@ private OperationInternal(OperationState<T> finalState)
             _stateLock = new AsyncLockWithValue<OperationState<T>>(finalState);
         }
 
-        public override Response RawResponse => _stateLock.TryGetValue(out var state) ? state.RawResponse : _rawResponse;
+        public override Response RawResponse => (_stateLock.TryGetValue(out var state) ? state.RawResponse : _rawResponse) ?? throw new InvalidOperationException("The operation does not have a response yet. Please call UpdateStatus or WaitForCompletion first.");
```

Comment:
Who calls this API?  If it's internal, would it make sense to provide a tester rather than throwing an exception?  It is not idiomatic to .NET to throw exceptions from properties - we should avoid doing this if possible.

Reply:
The case is, the operation is not complete yet, there is no response for the user to consume.
The user need to update the status till complete firstly.

Reply:
You are right, we should make this nullable instead and check nullable in the usage.

Reply:
Actually, we throw on incomplete operation [here](https://github.com/Azure/azure-sdk-for-net/blob/d49223f469c664319994ef18354c6e36e4325aca/sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs#L157) as well. To be consistent, we should throw on this as well.

Reply:
Get latest state during rehydration to avoid empty raw response, implemented in #42686